Mathlib/LinearAlgebra/BilinearMap.lean

```diff
@@ -204,6 +204,46 @@ theorem domRestrict₁₂_apply (f : M →ₛₗ[ρ₁₂] N →ₛₗ[σ₁₂]
     (x : p) (y : q) : f.domRestrict₁₂ p q x y = f x y := rfl
 #align linear_map.dom_restrict₁₂_apply LinearMap.domRestrict₁₂_apply
 
+section restrictScalars
+
+variable (R' S' : Type _)
+
+variable [Semiring R'] [Semiring S'] [Module R' M] [Module S' N] [Module R' Pₗ] [Module S' Pₗ]
+
+variable [SMulCommClass S' R' Pₗ] [SMulCommClass S R' Pₗ]
+
+variable [CompatibleSMul M (N →ₗ[S] Pₗ) R' R]
+
+variable [CompatibleSMul N Pₗ S' S]
```

Comment:
This is probably overkill; `IsScalarTower S' S N` and `IsScalarTower S' S Pₗ` would be easier to understand, and the only difference is `int`-bilinear maps over semirings which I doubt we care about.

Reply:
done